joinservice/internal/server/server.go

```diff
@@ -106,12 +116,35 @@ func (s *Server) IssueJoinTicket(ctx context.Context, req *joinproto.IssueJoinTi
 		return nil, status.Errorf(codes.Internal, "unable to generate Kubernetes join arguments: %s", err)
 	}
 
-	log.Infof("Querying K8sVersion ConfigMap")
+	log.Infof("Querying K8sVersion ConfigMap for Kubernetes version")
 	k8sVersion, err := s.getK8sVersion()
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "unable to get k8s version: %s", err)
 	}
 
+	log.Infof("Querying K8sVersion ConfigMap for components ConfigMap name")
+	componentsConfigMapName, err := s.getK8sComponentsConfigMapName()
+	if errors.Is(err, fs.ErrNotExist) {
+		// If the file does not exist, the Constellation was initialized with an version before 2.3.0
+		// As a migration step, the join service will create the ConfigMap with the K8s components which
+		// match the K8s minor version of the cluster.
+		log.Warnf("Reference to K8sVersion ConfigMap does not exist, creating fallback Components ConfigMap and referencing it in K8sVersion ConfigMap")
+		log.Warnf("This is expected if the Constellation was initialized with an CLI before version 2.3.0")
+		log.Warnf("DEPRECATION WARNING: This is a migration step and will be removed in a future release")
```

Comment:
```suggestion
		// If the file does not exist, the Constellation was initialized with a version before 2.3.0
		// As a migration step, the join service will create the ConfigMap with the K8s components which
		// match the K8s minor version of the cluster.
		log.Warnf("Reference to K8sVersion ConfigMap does not exist, creating fallback Components ConfigMap and referencing it in K8sVersion ConfigMap")
		log.Warnf("This is expected if the Constellation was initialized with a CLI before version 2.3.0")
		log.Warnf("DEPRECATION WARNING: This is a migration step and will be removed in a future release")
```
https://owl.purdue.edu/owl/general_writing/grammar/articles_a_versus_an.html